apps/server/src/services/rundown-service/delayedRundown.utils.ts

```diff
@@ -171,6 +171,43 @@ export async function cachedReorder(eventId: string, from: number, to: number) {
   return reorderedEvent;
 }
 
+/**
+ * Swaps two events
+ * @param {string} fromEventId
+ * @param {string} toEventId
+ */
+export async function cachedSwap(fromEventId: string, toEventId: string) {
+  const fromEventIndex = DataProvider.getIndexOf(fromEventId);
+  const toEventIndex = DataProvider.getIndexOf(toEventId);
+
+  if (fromEventIndex === -1 || toEventIndex === -1) {
+    invalidateFromError();
+    throw new Error('ID not found at index');
+  }
+
+  const rundownToUpdate = DataProvider.getRundown();
+
+  const fromEvent = rundownToUpdate.at(fromEventIndex) as OntimeEvent;
+  const toEvent = rundownToUpdate.at(toEventIndex) as OntimeEvent;
+
+  rundownToUpdate[fromEventIndex] = {
+    ...toEvent,
+    timeStart: fromEvent.timeStart,
+    timeEnd: fromEvent.timeEnd,
+    duration: fromEvent.duration,
+  };
+
+  rundownToUpdate[toEventIndex] = {
+    ...fromEvent,
+    timeStart: toEvent.timeStart,
+    timeEnd: toEvent.timeEnd,
+    duration: toEvent.duration,
+  };
```

Comment:
Since we have to do this 3 times 
1. in the optimistic mutation in react-query
2. here, to update the rundown in `DataProvider`
3. here again, to update the rundown in the `runtimeCacheStore`

Would it be best to write a small utility function in ontime-utils that handles this?
like 
``` ts
function swapData(rundown: OntimeRundown, eventA: OntimeEvent, eventB: OntimeEvent): OntimeRundown
```

This has the added benefit of facilitating maintenance of the feature, since the change would only happen in one place

Reply:
Resolving with https://github.com/cpvalente/ontime/pull/446/commits/617f107fb19e830ab5d5c051580465350f8fb3f3

The functions is used both in the frontend and backend (once for the `rundown` and `delayedRundown`)